crates/l2/sdk/src/calldata.rs

```diff
@@ -0,0 +1,314 @@
+                current_dynamic_offset += bytes_encoding.len();
+            }
+            Value::String(string_value) => {
+                write_u256(&mut ret, U256::from(current_dynamic_offset), current_offset)?;
+
+                let utf8_encoded = Bytes::copy_from_slice(string_value.as_bytes());
+                let bytes_encoding = encode_bytes(&utf8_encoded);
+                ret.extend_from_slice(&bytes_encoding);
+                current_dynamic_offset += bytes_encoding.len();
+            }
+            Value::Array(array_values) => {
+                write_u256(&mut ret, U256::from(current_dynamic_offset), current_offset)?;
+
+                let array_encoding = encode_array(array_values)?;
+                ret.extend_from_slice(&array_encoding);
+                current_dynamic_offset += array_encoding.len();
+            }
+            Value::Tuple(tuple_values) => {
+                if !is_dynamic(value) {
+                    let tuple_encoding = encode_tuple(tuple_values)?;
+                    ret.extend_from_slice(&tuple_encoding);
+                } else {
+                    write_u256(&mut ret, U256::from(current_dynamic_offset), current_offset)?;
+
+                    let tuple_encoding = encode_tuple(tuple_values)?;
+                    ret.extend_from_slice(&tuple_encoding);
+                    current_dynamic_offset += tuple_encoding.len();
+                }
+            }
+            Value::FixedArray(fixed_array_values) => {
+                if !is_dynamic(value) {
+                    let fixed_array_encoding = encode_tuple(fixed_array_values)?;
+                    ret.extend_from_slice(&fixed_array_encoding);
+                } else {
+                    write_u256(&mut ret, U256::from(current_dynamic_offset), current_offset)?;
+
+                    let tuple_encoding = encode_tuple(fixed_array_values)?;
+                    ret.extend_from_slice(&tuple_encoding);
+                    current_dynamic_offset += tuple_encoding.len();
+                }
+            }
+            Value::FixedBytes(bytes) => {
+                let mut to_copy = [0; 32];
+                to_copy.copy_from_slice(bytes);
+                copy_into(&mut ret, &to_copy, current_offset, 32)?;
+            }
+        }
+
+        current_offset += static_offset_value(value);
+    }
+
+    Ok(ret)
+}
+
+fn write_u256(values: &mut [u8], number: U256, offset: usize) -> Result<(), CalldataEncodeError> {
+    let mut to_copy = [0; 32];
+    number.to_big_endian(&mut to_copy);
+    copy_into(values, &to_copy, offset, 32)?;
+
+    Ok(())
+}
+
+// Returns the size that the value occupies in the static sector of the abi encoding.
+// For dynamic types, this is always 32 (the offset to the dynamic sector).
+// For static types, it's 32 unless the value is a static tuple or a fixed array, in which case
+// it's the sum of the sizes of their elements.
+fn static_offset_value(value: &Value) -> usize {
```

Comment:
Nit. Match arms that share the same behavior (without taking into account the contained value) can be nested with `|`.

Reply:
Addressed in [f5dd779](https://github.com/lambdaclass/ethrex/pull/1593/commits/f5dd77980ca64275dd85db14fdeeb15e0a3d8f1a)